src/bg_daemon/mac-update.sh

```diff
@@ -1,9 +1,14 @@
 #!/usr/bin/env bash
-osascript<<EOF
-tell application "Finder"
-    set desktop picture to POSIX file "$1"
-end tell
+if quack -h &>/dev/null; then
+    quack -i $1
+else
+    osascript<<EOF
+    tell application "Finder"
+        set desktop picture to POSIX file "$1"
+    end tell
 EOF
 
-# this is an ugly hack to force updating...
-killall Dock
+    # this is an ugly hack to force updating...
+    killall Dock
```

Comment:
What about the other solution (the background binary) did it work? I think it's cleaner than killall dock.